contracts/LivepeerProtocol.sol

```diff
@@ -523,11 +539,87 @@ contract LivepeerProtocol {
     }
 
     /*
-     * Pseudorandomly elect an active transcoder. Currently a placeholder
-     * TODO: take into account pricing, etc.
+     * Submit a transcoding job
+     * @param _streamId Unique stream identifier
+     * @param _transcodingOptions Output bitrates, formats, encodings
+     * @param _maxPricePerSegment Max price (in LPT base units) to pay for transcoding a segment of a stream
+     */
+    function job(uint256 _streamId, bytes32 _transcodingOptions, uint256 _maxPricePerSegment) returns (bool) {
+        address electedTranscoder = electCurrentActiveTranscoder(_maxPricePerSegment);
+
+        // Check if there is an elected current active transcoder
+        if (electedTranscoder == address(0)) throw;
+
+        jobs.push(Job(jobs.length - 1, _streamId, _transcodingOptions, _maxPricePerSegment, msg.sender, electedTranscoder, 0));
+
+        return true;
+    }
+
+    /*
+     * Computes the status of a job
+     * @param _jobId Job identifier
      */
-    function electCurrentActiveTranscoder() constant returns (address) {
-        return currentActiveTranscoders[uint(block.blockhash(block.number.sub(1))) % currentActiveTranscoders.length].id;
+    function jobStatus(uint256 _jobId) constant returns (JobStatus) {
+        if (jobs[_jobId].endBlock > 0 && jobs[_jobId].endBlock <= block.number) {
+            // A job is inactive if its end block is set and the current block is greater than or equal to the job's end block
+            return JobStatus.Inactive;
+        } else {
+            // A job is active if the current block is less than the job's termination block
+            return JobStatus.Active;
+        }
+    }
+
+    /*
+     * Return a job
+     * @param _jobId Job identifier
+     */
+    function getJob(uint256 _jobId) constant returns (uint256, uint256, bytes32, uint256, address, address, uint256) {
+        Job job = jobs[_jobId];
```

Comment:
I don't think we need to throw here because currently when `endJob()` is called, a job becomes inactive, but is not deleted from the `jobs` array. I lean toward keeping it this way because we retain a history of jobs, but curious to hear what you think. As a result, this function would never return an empty job and might also be used to inspect information about an inactive job.

I also realized that `jobs[_jobId]` will throw if `_jobId >= jobs.length` (handled by Solidity) and if `_jobId < 0` (since `_jobId` is a `uint256` any negative values will be rejected), so we don't need an additional check + throw for these cases.